src/schedule/dma_multi_chan_domain.c

```diff
@@ -303,11 +304,23 @@ static bool dma_multi_chan_domain_is_pending(struct ll_schedule_domain *domain,
 			    pipe_task->sched_comp)
 				continue;
 
-			/* it's too soon for this task */
-			if (!pipe_task->registrable &&
-			    pipe_task->task.start >
-			    platform_timer_get_noirq(timer_get()))
-				continue;
+			if (domain->full_sync) {
+				pdata = ll_sch_get_pdata(&pipe_task->task);
+				assert(pdata);
```

Comment:
This should be recoverable. i.e. we should emit some trace and return an error. I know we have assert() in other places, and we should remove, the idea is that we need to build in some survivability when errors do occur so that user space can tear down and rebuild, or developers have working IPC and can debug.

Reply:
I realize there is no need to check pdata here. This is done in schedule_task_init_ll().